charts/datahub/subcharts/datahub-gms/templates/deployment.yaml

```diff
@@ -94,12 +94,17 @@ spec:
             - name: DATAHUB_ANALYTICS_ENABLED
               value: "{{ .Values.global.datahub_analytics_enabled }}"
             - name: EBEAN_DATASOURCE_USERNAME
-              value: "{{ .Values.global.sql.datasource.username }}"
+              value: "{{ .Values.sql.datasource.username | default .Values.global.sql.datasource.username }}"
```

Comment:
The templating will attempt to use the deployment specific config but will fallback to the existing global config if not found for backwards compatibility.